app/Models/UserQuery.php

```diff
@@ -0,0 +1,218 @@
+<?php
+
+/**
+ * Contains the description of a user query
+ * 
+ * It allows to extract the meaningful bits of the query to be manipulated in an
+ * easy way.
+ */
+class FreshRSS_UserQuery {
+
+	private $deprecated = false;
+	private $get;
+	private $get_name;
+	private $get_type;
+	private $name;
+	private $order;
+	private $search;
+	private $state;
+	private $url;
+	private $feed_dao;
+	private $category_dao;
+
+	/**
+	 * @param array $query
+	 * @param FreshRSS_Searchable $feed_dao
+	 * @param FreshRSS_Searchable $category_dao
+	 */
+	public function __construct($query, FreshRSS_Searchable $feed_dao = null, FreshRSS_Searchable $category_dao = null) {
+		$this->category_dao = $category_dao;
+		$this->feed_dao = $feed_dao;
+		if (isset($query['get'])) {
+			$this->parseGet($query['get']);
+		}
+		if (isset($query['name'])) {
+			$this->name = $query['name'];
```

Comment:
Name should be required, isn't it? In this case `isset` test should be facultative. But it's ok like that too ;)